src/components/accordion/use-div-height.ts

```diff
@@ -0,0 +1,31 @@
+import { useEffect, useRef, useState } from 'react';
+
+/**
+ * Lovely height measurement, from here:
+ * https://github.com/raunofreiberg/reach-ui/blob/6742f0cdcef05ba23077f63c21e3ede3bbfcae52/packages/accordion/examples/animated.example.js#L19-L39
+ */
+export function useDivHeight() {
+  const ref = useRef(null);
+  const [height, setHeight] = useState(0);
+
+  useEffect(() => {
+    const resizeObserver = new ResizeObserver(([entry]) => {
+      requestAnimationFrame(() => {
+        if (!entry) {
+          return;
+        }
+        setHeight(entry.target.getBoundingClientRect().height);
+      });
+    });
+
+    if (ref.current) {
+      resizeObserver.observe(ref.current);
+    }
+
+    return () => {
+      resizeObserver.disconnect();
+    };
+  }, []);
+
+  return { ref, height };
+}
```

Comment:
I always need some time to understand custom hooks. I never can make any sense out of them on the first sight. I really love this one to always get the current height of a component 🌟